src/libcore/mem.rs

```diff
@@ -448,17 +448,29 @@ pub unsafe fn uninitialized<T>() -> T {
 pub fn swap<T>(x: &mut T, y: &mut T) {
     unsafe {
         // Give ourselves some scratch space to work with
-        let mut t: T = uninitialized();
+        let mut t: [u8; 16] = uninitialized();
```

Comment:
Minor point: Instead of using a literal `16` here, it might be slightly better to stick a

```rust
const SWAP_BLOCK_SIZE: usize = 16;
```

at the top of the function. This would make it slightly less error-prone to tweak the value later, and it would make it easy to, e.g., pick different sizes for different architectures.